core/src/main/scala/kafka/cluster/Replica.scala

```diff
@@ -50,12 +43,25 @@ class Replica(val brokerId: Int,
   // the LEO of leader at time t. This is used to determine the lag of this follower and ISR of this partition.
   @volatile private[this] var _lastCaughtUpTimeMs = 0L
 
-  def isLocal: Boolean = log.isDefined
+  def logStartOffset: Long =
+    _logStartOffset
+
+  private def logStartOffset_=(newLogStartOffset: Long) {
+    _logStartOffset = newLogStartOffset
+    trace(s"Setting log start offset for remote replica $brokerId for partition $topicPartition to [$newLogStartOffset]")
+  }
+
+  def logEndOffsetMetadata: LogOffsetMetadata =
+    _logEndOffsetMetadata
+
+  def logEndOffset: Long = logEndOffsetMetadata.messageOffset
 
   def lastCaughtUpTimeMs: Long = _lastCaughtUpTimeMs
 
-  info(s"Replica loaded for partition $topicPartition with initial high watermark $initialHighWatermarkValue")
-  log.foreach(_.onHighWatermarkIncremented(initialHighWatermarkValue))
+  def lastCaughtUpTimeMs_=(lastCaughtUpTimeMs: Long) {
+    _lastCaughtUpTimeMs = lastCaughtUpTimeMs
+    trace(s"Setting log caught offset time for replica $brokerId, partition $topicPartition to $lastCaughtUpTimeMs")
```

Comment:
nit: "log caught offset time" -> "last caught up time"?

Reply:
Cleaned up this class a bit more and streamlined logging.